src/test/java/hudson/plugins/tics/AbstractApiCallTest.java

```diff
@@ -0,0 +1,91 @@
+package hudson.plugins.tics;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.junit.Test;
+
+public class AbstractApiCallTest {
+
+    private class NoProxyTestCase {
+        public List<Pattern> noProxyPatterns;
+        public String targetUrl;
+        public boolean expectedResult;
+
+        public NoProxyTestCase(final List<Pattern> noProxyPatterns, final String targetUrl, final boolean expectedResult) {
+            this.noProxyPatterns = noProxyPatterns;
+            this.targetUrl = targetUrl;
+            this.expectedResult = expectedResult;
+        }
+    }
+
+    private List<Pattern> getPatterns(final List<String> regexes) {
+        final List<Pattern> noProxyPatterns = new ArrayList<>();
```

Comment:
Can be written simpler, more functional, as `return regexes.stream().map(Pattern::compile).collect(Collectors.toList());`

Reply:
Fixed in https://github.com/jenkinsci/tics-plugin/pull/28/commits/6139d5dc4e4d65e14756fd10a5e6f4192e51e731